clang/lib/CodeGen/CGLoopInfo.cpp

```diff
@@ -706,8 +706,13 @@ void LoopInfoStack::push(BasicBlock *Header, clang::ASTContext &Ctx,
     // 1 - disable unroll.
     // other positive integer n - unroll by n.
     if (OpenCLHint || UnrollHint) {
-      ValueInt = OpenCLHint ? OpenCLHint->getUnrollHint()
-                            : UnrollHint->getUnrollHint();
+      Expr *ValueExpr = nullptr;
+      ValueInt =
```

Comment:
Can this be simplified to an if-else please?  The nesting of conditionals is pretty unreadable particularly when mixed with the variable assignment.

Also, what happens when BOTH hints are true?  Does that need asserting, or some sort of 'min' behavior?

Reply:
You can see how 'if-else' approach looked in the first commit of this PR. I find current 'ternary' approach more laconic and good-looking. However, it may be written somehow better.

These two hints cannot be both true, as the first one is supported only in OpenCL and the second one is only in SYCL

Reply:
if-else is going to be better here... having ValueExpr's value change in the ternary is just odd.

I'd expect an assert then for !OpenCLHint || !UnrolLHint to mkae that clear.

Reply:
Even if these both attributes are present, they are handled separately in a loop (line 688). It is not possible that one attribute is successfully cast to two or three different attributes (initialization in 689-692). So, I expect no assert here.